ctapipe/tools/muon_reconstruction.py

```diff
@@ -5,121 +5,176 @@
+            return
+
+        # intensity_fitter does not support a mask yet, set ignored pixels to 0
+        image[~mask] = 0
+
+        result = self.intensity_fitter(
+            tel_id,
+            ring.ring_center_x,
+            ring.ring_center_y,
+            ring.ring_radius,
+            image,
+            pedestal=1.1,
+        )
 
-        numev = 0
-        self.num_muons_found = defaultdict(int)
-
-        for event in tqdm(self.source, desc='detecting muons'):
-
-            self.calib(event)
-            muon_evt = analyze_muon_event(event)
-
-            if numev == 0:
-                _exclude_some_columns(event.inst.subarray, self.writer)
-
-            numev += 1
-
-            if not muon_evt['MuonIntensityParams']:
-                # No telescopes  contained a good muon
-                continue
-            else:
-                if self.display:
-                    plot_muon_event(event, muon_evt)
-
-                for tel_id in muon_evt['TelIds']:
-                    idx = muon_evt['TelIds'].index(tel_id)
-                    intens_params = muon_evt['MuonIntensityParams'][idx]
+        self.log.info(
+            f'Muon fit: r={ring.ring_radius:.2f}'
+            f', width={result.ring_width:.4f}'
+            f', efficiency={result.optical_efficiency_muon:.2%}',
+        )
 
-                    if intens_params is not None:
-                        ring_params = muon_evt['MuonRingParams'][idx]
-                        cam_id = str(event.inst.subarray.tel[tel_id].camera)
-                        self.num_muons_found[cam_id] += 1
-                        self.log.debug("INTENSITY: %s", intens_params)
-                        self.log.debug("RING: %s", ring_params)
-                        self.writer.write(table_name=cam_id,
-                                          containers=[intens_params,
-                                                      ring_params])
+        if self.outfile:
+            self.writer.write(f'tel_{tel_id}', [event_index, ring, result])
 
-                self.log.info(
-                    "Event Number: %d, found %s muons",
-                    numev, dict(self.num_muons_found)
-                )
+    def pixel_to_telescope_frame(self, tel_id):
+        telescope = self.source.subarray.tel[tel_id]
+        cam = telescope.camera.geometry
+        camera_frame = CameraFrame(
+            focal_length=telescope.optics.equivalent_focal_length,
+            rotation=cam.cam_rotation,
+        )
+        cam_coords = SkyCoord(x=cam.pix_x, y=cam.pix_y, frame=camera_frame)
+        return cam_coords.transform_to(TelescopeFrame())
 
     def finish(self):
-        Provenance().add_output_file(self.outfile,
-                                     role='dl1.tel.evt.muon')
-        self.writer.close()
+        if self.outfile:
+            Provenance().add_output_file(
+                self.outfile,
+                role='dl1.tel.evt.muon',
```

Comment:
This naming has now changed, and since this isn't in a DL1 file, maybe just set the role to "telescope muon analysis parameters". It's just some text to recall why the file was produced semantically, so so far there is no dictionary for what to put there.